src/pages/trader-tools/common/_utility.js

```diff
@@ -23,42 +23,155 @@ const rawCalculation = (values, specialFormula) => {
     return formula
 }
 
-const getMargin = (values) => {
+export const getMargin = (values) => {
     const { volume, assetPrice, leverage, contractSize } = values
     const specialFormula = (volume * contractSize * assetPrice) / leverage.name
     let margin_formula = rawCalculation(values, specialFormula)
     return toFixed(margin_formula)
 }
 
-const getPipValue = (values) => {
+export const getPipValue = (values) => {
     const { volume, pointValue, contractSize } = values
     const specialFormula = volume * contractSize * pointValue
     let pip_formula = rawCalculation(values, specialFormula)
 
     return toFixed(pip_formula)
 }
 
-const getSwapChargeSynthetic = (values) => {
+export const getSwapChargeSynthetic = (values) => {
     const { volume, assetPrice, swapRate, contractSize } = values
-    const specialFormula = swap_formula_synthetic = (volume * contractSize * assetPrice * (swapRate / 100)) / 360
+    const specialFormula = (swap_formula_synthetic =
+        (volume * contractSize * assetPrice * (swapRate / 100)) / 360)
     let swap_formula_synthetic = rawCalculation(values, specialFormula)
 
     return toFixed(swap_formula_synthetic)
 }
 
-const getSwapChargeForex = (values) => {
+export const getSwapChargeForex = (values) => {
     const { volume, pointValue, swapRate, contractSize } = values
     const swap_formula_forex = volume * contractSize * pointValue * swapRate
     return toFixed(swap_formula_forex)
 }
 
-const toFixed = (val) => {
+// PnL Margin Calculator
+export const getPnlMarginCommon = (values, action) => {
+    let {
+        assetPrice,
+        stopLossAmount,
+        takeProfitAmount,
+        volume,
+        contractSize,
+        stopLossLevel,
+        takeProfitLevel,
+        pointValue,
+    } = values
+    assetPrice = Number(assetPrice)
+    stopLossAmount = Number(stopLossAmount)
+    takeProfitAmount = Number(takeProfitAmount)
+    stopLossLevel = Number(stopLossLevel)
+    takeProfitLevel = Number(takeProfitLevel)
+
+    if (action === 'getStopLossLevel') {
```

Comment:
these are already good, however using switch might be better if it's only single if ( not mandatory )